eng/tools/azure-sdk-tools/azpysdk/sphinx.py

```diff
@@ -220,22 +255,25 @@ def run(self, args: argparse.Namespace) -> int:
             logger.info(f"Running sphinx against {package_name}")
 
             # prep env for sphinx
-            doc_folder = os.path.join(staging_directory, "docgen")
             site_folder = os.path.join(package_dir, "website")
+            doc_folder = os.path.join(staging_directory, "docgen")
 
             if should_build_docs(package_name):
```

Comment:
let's move this `should_build_docs` to a precheck above this entire section of code.

We check multiple places whether or not we should build docs, and if we shouldn't we keep having to check this. I recognize you probably pulled this from previous code state, but if we're here we might as well improve it.